### PR TITLE
Initialization logic is rewritten

### PR DESCRIPTION
**Pull request by jonano614**
- xdag_init function is splitted to logical parts
- some global varibales are moved to global.h file
- variables g_is_miner, g_is_pool, g_xdag_pool and g_light_mode are replaces with one variable g_xdag_type
- dnet_crypt.c file is formatted, some reduntunt lines are remove
- reading dnet key is moved to separate function
- preparations for adding single wallet file "xdag.wallet"
- windows-compilation and formatting for PR: https://github.com/XDagger/xdag/pull/461

### CODE REVIEW

File: client/block.c [hunk 1]
```diff
@@ -839,7 +839,7 @@ void *add_block_callback(void *block, void *data)
 
 	pthread_mutex_unlock(&block_mutex);
 
-	if(res >= 0) {
+	if(res >= 0 && g_xdag_type == XDAG_POOL) {
```
Comment (jonano614) [inline client/block.c]: Blocks synchronization logic is initialized only for pools now, so xdag_sync_pop_block can be called only for pools

File: client/block.c [hunk 2]
```diff
@@ -1194,15 +1194,6 @@ static void *work_thread(void *arg)
 			} else if (time(NULL) - last_time_nmain_unequal > MAX_TIME_NMAIN_STALLED) {
 				g_block_production_on = 1;
 			}
-
-			if (g_block_production_on) {
```
Comment (xrdavies) [inline client/block.c]: Do we need remove these logs?

File: client/commands.c [hunk 3]
```diff
@@ -331,7 +332,7 @@ int xdag_command(char *cmd, FILE *out)
 
 	XDAG_COMMAND *command = find_xdag_command(cmd);
 
-	if(!command || (command->avaibility == 1 && !g_is_miner) || (command->avaibility == 2 && g_is_miner)) {
+	if(!command || (command->avaibility == 1 && g_xdag_type == XDAG_POOL) || (command->avaibility == 2 && g_xdag_type == XDAG_WALLET)) {
```
Comment (xrdavies) [inline client/commands.c]: Maybe just use two functions
isPool() { 
return g_xdag_type == XDAG_POOL;
}

isWallet() {
return g_xdag_type == XDAG_WALLET;
}

File: client/global.c [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+int g_xdag_state = XDAG_STATE_INIT;
+int g_xdag_testnet = 0;
+int g_xdag_run = 0;
+enum xdag_field_type g_block_header_type = XDAG_FIELD_HEAD;
+struct xdag_stats g_xdag_stats;
+struct xdag_ext_stats g_xdag_extstats;
+int g_disable_mining = 0;
+enum xdag_type g_xdag_type = 0;
+char *g_coinname, *g_progname;
```
Comment (xrdavies) [inline client/global.c]: Emmm, there is one extra character on windows.

File: client/terminal.c [hunk 5]
```diff
@@ -38,6 +38,7 @@ int terminal(void)
 
 	if(!xdag_network_init()) {
 		printf("Can't initialize sockets");
+		return -1;
```
Comment (xrdavies) [inline client/terminal.c]: Good point!!

File: client/xdag_config.c [hunk 6]
```diff
@@ -13,111 +11,93 @@
...
-            end--;
-        }
-        if(end - start != len)
-        {
-            strncpy(str, start, end-start);
-            str[end-start] = '\0';
-        }
-    }
-    return str;
+	if(str != NULL) {
```
Comment (xrdavies) [inline client/xdag_config.c]: trim can be moved to utils, or we can introduce a file strutils.c to hold all related functions.

File: client/global.c (hunk 4 above)
Comment (jonano614) [reply client/global.c]: Sorry, I didn't understand what is wrong here.

File: client/commands.c (hunk 3 above)
Comment (jonano614) [reply client/commands.c]: Done

File: client/block.c (hunk 2 above)
Comment (jonano614) [reply client/block.c]: Done.

File: client/global.h [hunk 7]
```diff
@@ -66,4 +66,7 @@ extern int g_disable_mining;
 //Test network and main network have different types of the block headers, so blocks from different networks are incompatible
 extern enum xdag_field_type g_block_header_type;
 
+inline int is_pool() { return g_xdag_type == XDAG_POOL; }
+inline int is_wallet() { return g_xdag_type == XDAG_WALLET; }
```
Comment (sgaragagghu) [inline client/global.h]: i think you are missing void :)

Comment (jonano614) [reply client/global.h]: Ah, yes.

**Pull request merged**